### PR TITLE
Catch uncaught opener errors

### LINKED ISSUE #381: openAnalyzer causes crash when run on a server
#### Issue description

The default settings of the plugin have `openAnalyzer: true`, but when I have that set to default on my server, it causes an immediate crash:

```
Error: spawn ENOTDIR
    at ChildProcess.spawn (internal/child_process.js:407:11)
    at spawn (child_process.js:548:9)
    at Object.execFile (child_process.js:232:17)
    at opener (/home/coder/c/customer/site/node_modules/opener/lib/opener.js:65:25)
    at Server.<anonymous> (/home/coder/c/customer/site/node_modules/webpack-bundle-analyzer/lib/viewer.js:96:11)
```

Since the opener is pretty non-critical, it'd be nice to just wrap it in a try / catch and log or no-op if it throws.

#### Technical info

<pre>
  System:
    OS: Linux 4.19 Arch Linux
    CPU: (16) x64 Intel(R) Xeon(R) CPU @ 2.30GHz
    Memory: 13.44 GB / 58.98 GB
    Container: Yes
    Shell: 5.0.18 - /bin/bash
  Binaries:
    Node: 12.16.0 - ~/.nvm/versions/node/v12.16.0/bin/node
    Yarn: 1.22.4 - ~/.yarn/bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v12.16.0/bin/npm
</pre>

#### Debug info
How do you use this module? **Webpack plugin**

If plugin, what options were provided? **`new BundleAnalyzerPlugin()`**

### PR DESCRIPTION
**Pull request by wbobeirne**
Closes #381.

### CODE REVIEW

**Comment by valscion** [comment]
Please sign CLA and fix CI ☺️

File: src/viewer.js [hunk 1]
```diff
@@ -229,3 +229,14 @@ function getChartData(analyzerOpts, ...args) {
...
 }
+
+/**
+ * Calls opener on a URI, but silently try / catches it.
+ */
+function safeOpener(uri) {
+  try {
+    opener(uri)
+  } catch (err) {
+    logger.debug(`Opener failed to open "${uri}":\n${err}`)
```
Comment (th0r) [inline src/viewer.js:240]: `logger` is not defined.

File: src/viewer.js [hunk 2]
```diff
@@ -229,3 +229,14 @@ function getChartData(analyzerOpts, ...args) {
 
   return chartData;
 }
+
+/**
+ * Calls opener on a URI, but silently try / catches it.
+ */
+function safeOpener(uri) {
```
Comment (th0r) [inline src/viewer.js:236]: Let's better move this function to `utils.js`.

**Review by th0r — CHANGES_REQUESTED** [review]

File: package-lock.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "webpack-bundle-analyzer",
-  "version": "3.8.0",
+  "version": "3.9.0",
```
Comment (th0r) [inline package-lock.json:3]: Don't change the version in this PR please.

File: src/utils.js [hunk 4]
```diff
@@ -51,3 +52,14 @@ exports.defaultTitle = function () {
 
   return `${process.env.npm_package_name || 'Webpack Bundle Analyzer'} [${currentTime}]`;
 };
+
+/**
+ * Calls opener on a URI, but silently try / catches it.
+ */
+exports.safeOpener = function (uri, logger) {
```
Comment (th0r) [inline src/utils.js:59]: Let's better rename it to just `open` please.

**Review by th0r — CHANGES_REQUESTED** [review]

**Review by th0r — APPROVED** [review]

**Comment by th0r** [comment]
Thanks!

**Pull request merged**